features/prog.go

```diff
@@ -145,6 +160,89 @@ func haveProgType(pt ebpf.ProgramType) error {
 	return err
 }
 
+// HaveProgHelper probes the running kernel for the availability of the specified helper
+// function to a specified program type.
+// Return values have the following semantics:
+//
+//   err == nil: The feature is available.
+//   errors.Is(err, ebpf.ErrNotSupported): The feature is not available.
+//   err != nil: Any errors encountered during probe execution, wrapped.
+//
+// Note that the latter case may include false negatives, and that program creation may
+// succeed despite an error being returned.
+// Only `nil` and `ebpf.ErrNotSupported` are conclusive.
+//
+// Probe results are cached and persist throughout any process capability changes.
+func HaveProgHelper(pt ebpf.ProgramType, helper asm.BuiltinFunc) error {
+	if err := validateProgType(pt); err != nil {
+		return err
+	}
+
+	if err := validateProgHelper(helper); err != nil {
+		return err
+	}
+
+	return haveProgHelper(pt, helper)
+}
+
+func validateProgHelper(helper asm.BuiltinFunc) error {
+	if helper > helper.Max() {
+		return os.ErrInvalid
+	}
+
+	return nil
+}
+
+func haveProgHelper(pt ebpf.ProgramType, helper asm.BuiltinFunc) error {
+	pc.helperMu.Lock()
+	defer pc.helperMu.Unlock()
+	err, ok := pc.progHelpers[pt][helper]
+	if ok {
```

Comment:
To indicate the short lifespan of `err`:

```suggestion
	if err, ok := pc.progHelpers[pt][helper]; ok {
```